### PR TITLE
Use fixture instead of deprecated yield_fixture

### LINKED ISSUE #3351: backend: @pytest.yield_fixture is deprecated.
When building copr-backend on my F40 I see:

```
tests/test_frontend.py:14
  /builddir/build/BUILD/copr-backend-git-39.f91d7fc/tests/test_frontend.py:14: PytestDeprecationWarning: @pytest.yield_fixture is deprecated.
  Use @pytest.fixture instead; they are the same.
    @pytest.yield_fixture
```
We should fix this before it is removed.

### PR DESCRIPTION
**Pull request by FrostyX**
Fix #3351

The deprecation warning even says "Use @pytest.fixture instead; they are the same".

### CODE REVIEW

**Comment by FrostyX** [comment]
The failure is unrelated to this PR but it is caused by #3365.
But to be sure, let's wait until it's green.

**Review by nikromen — APPROVED** [review]
seems straightforward but yup, let's wait for the CI to be green before merging

**Comment by FrostyX** [comment]
CI fixed, PTAL.

**Pull request merged**
